pkg/docker/docker.go

```diff
@@ -592,14 +594,23 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) (err error) {
 		config.AttachStdout = true
 	}
 
-	glog.V(2).Infof("Creating container using config: %+v", config)
 	ccopts := docker.CreateContainerOptions{Name: "", Config: &config}
+	ccopts.HostConfig = &docker.HostConfig{}
 	if opts.TargetImage {
-		ccopts.HostConfig = &docker.HostConfig{PublishAllPorts: true, NetworkMode: opts.NetworkMode}
+		ccopts.HostConfig.PublishAllPorts = true
+		ccopts.HostConfig.NetworkMode = opts.NetworkMode
 	} else if opts.NetworkMode != "" {
-		ccopts.HostConfig = &docker.HostConfig{NetworkMode: opts.NetworkMode}
+		ccopts.HostConfig.NetworkMode = opts.NetworkMode
 	}
 
+	if opts.CGroupLimits != nil {
+		ccopts.HostConfig.Memory = opts.CGroupLimits.MemoryLimitBytes
+		ccopts.HostConfig.MemorySwap = opts.CGroupLimits.MemorySwap
```

Comment:
Shouldn't this be "always -1"? Are we going to rely on all callers to set this to -1?